### ISSUE TITLE
Write-RegEx.ps1 help example #4 doesn't work for all email addresses

### ISSUE THREAD

**Issue by jon646** [StartAutomating/Irregular#39]
In Write-RegEx.ps1, the fourth .Example provides code for capturing an e-mail address, but the sample code provided fails if the email address contains subdomains (e.g. joe@nyu.gmail.com or boris@pm.gov.uk).  

I'm also not sure what would happen if an email address or domain contains multiple hyphens (mario-the-plumber@nyu.gmail.com or barnum@show-of-shows.net).

FWIW, emailRegex.com has a pretty exhaustive .NET RegEx string for email addresses, but it might be too much for an example.

**To Reproduce**
emailRegex.Match("joe@nyu.gmail.com")

**Expected behavior**
Returns the entire email address

**Actual behavior**
Returns joe@nyu.gmail

**Additional context**
This came up during your talk to the NY PowerShell MeetUp.

Cool talk, btw!

**Comment by StartAutomating** [comment]
Thanks for pointing this out!

The RegEx has been updated in a test branch along with a .regex.source.ps1 file and some additional documentation.

I took at look at the emailregex site and it looks like significant poorly managed overkill.

The goal with most of these Regexes isn't to be 100% perfect according to some standard, but to be "close enough" for most work.  To that end there are some things that I know are valid in email addresses that I have no intention of supporting (e.g. foo @127.0.0.1 ).

Subdomains are a much more common scenario.

This fix should be integrated into the next full build.
